File: pbmx_ffi/java/io/rmf/pbmx/PublicKey.java
package io.rmf.pbmx;

import io.rmf.pbmx.ffi.LibPbmx;
import com.sun.jna.Pointer;

public final class PublicKey {
    PublicKey(Pointer handle) {
        this.handle = handle;
    }

    public PublicKey(PrivateKey sk) {
        this.handle = LibPbmx.INSTANCE.pbmx_public_key(sk.handle);
    }

    public Fingerprint fingerprint() {
        return new Fingerprint(LibPbmx.INSTANCE.pbmx_key_fingerprint(this.handle));
    }

    @Override
    public void finalize() {
        LibPbmx.INSTANCE.pbmx_delete_public_key(this.handle);
    }

    Pointer handle;
}

File: pbmx_droid/app/src/main/java/io/rmf/pbmx/PublicKey.kt
package io.rmf.pbmx

import com.sun.jna.Pointer
import io.rmf.pbmx.ffi.FFI

class PublicKey {

    internal var handle: Pointer

    internal constructor(handle: Pointer) {
        this.handle = handle
    }

    constructor(sk: PrivateKey) {
        this.handle = FFI.pbmx_public_key(sk.handle)
    }

    val fingerprint get() = Fingerprint(FFI.pbmx_key_fingerprint(this.handle))

    fun finalize() {
        FFI.pbmx_delete_public_key(this.handle)
    }
}
